st2common/bin/st2-run-pack-tests

```diff
@@ -273,7 +285,66 @@ fi
 echo "Running tests..."
 # Note: We run nosetests with "--exe" option so it also runs test files which are executable
 # (pack install command automatically makes all the files, including test files executable)
-nosetests -s -v --exe ${PACK_TESTS_PATH}
+NOSE_OPTS=(-s -v --exe --rednose --immediate)
+
+# Should we pass in arguments to enable test coverage
+if [ "${ENABLE_COVERAGE}" = true ]; then
+  verbose_log "Enabling nosetests coverage"
+
+  # base options to enable test coverage printouts
+  NOSE_OPTS+=(--with-coverage)
+
+  # Now, by default nosetests prints out test coverage for every module found
+  # in the $PYTHONPATH... as you can imagine this is not ideal and may contain
+  # a LOT of files. To only print out the modules that we care about we need to
+  # specify them on the cdmline using --cover-package=package.name
+
+  # There are two directories that way contain python files
+  PACK_PYTHON_DIRS=("${PACK_PATH}/actions/" "${PACK_PATH}/sensors/")
+
+  # There are two scenarios for python files in the PACK_PYTHON_DIRS
+  # 1) there are xxx.py files in the root of the directory (ex: actions/example.py)
+  # 2) there are sub directories that contain __init__.py files that signify
+  #    to python that they're a module (ex: actions/lib/__init.py). All files
+  #    within this sub directory will be checked for coverage if we
+  #    specify the option for the base module (ex: --cover-package=lib)
+
+  # for each of the directories in the pack that may contain python files
+  for pack_python_dir in ${PACK_PYTHON_DIRS}; do
+
+    # find all *.py files in the root of the $pack_python_dir
+    for py_absolute in $(find $pack_python_dir  -mindepth 1 -maxdepth 1 -name "*.py"); do
```

Comment:
After some experimentation and reading the nosetest/coverage code i've been able to greatly simplify this logic.

Now, it simply finds all sub-directories within the pack (excluding the "tests" directory) and explicitly includes those for reporting. This successfully hides the erroneous imports, simplifies the code, and should make it more future-proof.